app/keepers/keepers.go

```diff
@@ -257,6 +262,26 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 		AddRoute(ibctransfertypes.ModuleName, appKeepers.TransferStack)
 	// Note: the sealing is done after creating wasmd and wiring that up
 
+	// ICQ Keeper
+	icqKeeper := icqkeeper.NewKeeper(
+		appCodec,
+		appKeepers.keys[icqtypes.StoreKey],
+		appKeepers.GetSubspace(icqtypes.ModuleName),
+		appKeepers.IBCKeeper.ChannelKeeper, // may be replaced with middleware
+		appKeepers.IBCKeeper.ChannelKeeper,
+		&appKeepers.IBCKeeper.PortKeeper,
+		appKeepers.ScopedICQKeeper,
+		bApp,
```

Comment:
Uhhhh, why does this take in baseApp -- we may have to audit this and all dep updates carefully then. Its not safe to just give base app access to modules.

Reply:
it needs a Querier. I was wondering if there was a something more restrictive that we could pass there.

Reply:
Gotcha, I think its worth making a wrapper struct on our side for now then